api/v1alpha3/awsmachine_webhook.go

```diff
@@ -38,12 +39,32 @@ func (r *AWSMachine) SetupWebhookWithManager(mgr ctrl.Manager) error {
 }
 
 // +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha3-awsmachine,mutating=false,failurePolicy=fail,groups=infrastructure.cluster.x-k8s.io,resources=awsmachines,versions=v1alpha3,name=validation.awsmachine.infrastructure.cluster.x-k8s.io
+// +kubebuilder:webhook:verbs=create;update,path=/mutate-infrastructure-cluster-x-k8s-io-v1alpha3-awsmachine,mutating=true,failurePolicy=fail,groups=infrastructure.cluster.x-k8s.io,resources=awsmachines,versions=v1alpha3,name=default.awsmachine.infrastructure.cluster.x-k8s.io
 
 var _ webhook.Validator = &AWSMachine{}
+var _ webhook.Defaulter = &AWSMachine{}
+
+// Default implements webhook.Defaulter so a mutating webhook will be registered for the type
+func (r *AWSMachine) Default() {
+	if r.ObjectMeta.Labels == nil {
+		return
+	}
+	if _, ok := r.ObjectMeta.Labels[clusterv1.MachineControlPlaneLabelName]; !ok {
+		return
+	}
+	if r.Spec.CloudInit == nil || !r.Spec.CloudInit.Encrypted {
```

Comment:
I assume the answer is "no" but would someone ever create/update an AWSMachine with the SecretARN set but Encrypted false? If so, we'd lose the SecretARN here.